src/Interpreters/parseDatabaseAndTableName.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <Common/Exception.h>
+#include <base/defines.h>
+#include <base/types.h>
+#include <boost/algorithm/string.hpp>
+
+namespace DB
+{
+    template <typename Query>
+    inline std::pair<String, String> parseDatabaseAndTableName(const Query & query, const String & current_database)
+    {
+        String database;
+        String table;
+
+        if (query.from_table.contains("."))
```

Comment:
It looks very strange.
The AST should be already structured, with the database and table name in separate child nodes, at least under the CompoundIdentifier.